test/sagemaker_tests/huggingface_pytorch/training/integration/local/test_single_machine_training.py

```diff
@@ -0,0 +1,92 @@
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import os
+
+import pytest
+import sagemaker.huggingface
+from sagemaker.huggingface import HuggingFace
+import botocore
+from datasets.filesystems import S3FileSystem
+
+from ...utils.local_mode_utils import assert_files_exist
+from ...integration import ROLE, distrilbert_script
+
+
+@pytest.mark.model("hf_bert")
+@pytest.mark.integration("hf_local")
+@pytest.mark.skip_py2_containers
+def test_distilbert_base(docker_image, processor, instance_type, sagemaker_local_session, tmpdir):
+    from datasets import load_dataset
+    from transformers import AutoTokenizer
+
+    # tokenizer used in preprocessing
+    tokenizer_name = 'distilbert-base-uncased'
+
+    # dataset used
+    dataset_name = 'imdb'
+
+    # s3 key prefix for the data
+    s3_prefix = 'samples/datasets/imdb'
+    # load dataset
+    dataset = load_dataset(dataset_name)
+
+    # download tokenizer
+    tokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
+
+    # tokenizer helper function
+    def tokenize(batch):
+        return tokenizer(batch['text'], padding='max_length', truncation=True)
+
+    # load dataset
+    train_dataset, test_dataset = load_dataset('imdb', split=['train', 'test'])
+    test_dataset = test_dataset.shuffle().select(range(100))  # smaller the size for test dataset to 10k
+
+    # tokenize dataset
+    train_dataset = train_dataset.map(tokenize, batched=True, batch_size=len(train_dataset))
+    test_dataset = test_dataset.map(tokenize, batched=True, batch_size=len(test_dataset))
+
+    # set format for pytorch
+    train_dataset.rename_column_("label", "labels")
+    train_dataset.set_format('torch', columns=['input_ids', 'attention_mask', 'labels'])
+    test_dataset.rename_column_("label", "labels")
+    test_dataset.set_format('torch', columns=['input_ids', 'attention_mask', 'labels'])
+
+    # hyperparameters, which are passed into the training job
+    hyperparameters = {'epochs': 1,
+                       'train_batch_size': 32,
+                       'model_name': 'distilbert-base-uncased'
+                       }
+
+    s3 = S3FileSystem()
+
+    # save train_dataset to s3
+    training_input_path = f's3://{sagemaker_local_session.default_bucket()}/{s3_prefix}/train'
+    train_dataset.save_to_disk(training_input_path, fs=s3)
+
+    # save test_dataset to s3
+    test_input_path = f's3://{sagemaker_local_session.default_bucket()}/{s3_prefix}/test'
+    test_dataset.save_to_disk(test_input_path, fs=s3)
+
+    estimator = HuggingFace(entry_point=distrilbert_script,
+                            instance_type='ml.p3.16xlarge',
+                            sagemaker_session=sagemaker_local_session,
+                            instance_count=1,
+                            role=ROLE,
+                            transformers_version='4.4',
+                            pytorch_version='1.6',
```

Comment:
That should be assigned based on command line parameters. 
And why do you use short version? If we have 1.6.0, 1.6.1, 1.6.2, how will the sdk know which version to pick?

Reply:
We will just provide image_uri and won't be passing these versions